plugins/catalog-backend/src/ingestion/processors/AwsOrganizationCloudAccountProcessor.ts

```diff
@@ -33,9 +39,40 @@ const ORGANIZATION_ANNOTATION: string = 'amazonaws.com/organization-id';
  * If custom authentication is needed, it can be achieved by configuring the global AWS.credentials object.
  */
 export class AwsOrganizationCloudAccountProcessor implements CatalogProcessor {
+  logger: Logger;
   organizations: Organizations;
-  constructor() {
+  providers: AwsOrganizationProviderConfig[];
+
+  static fromConfig(config: Config, options: { logger: Logger }) {
+    const c = config.getOptionalConfig('catalog.processors.awsOrganization');
+    return new AwsOrganizationCloudAccountProcessor({
+      ...options,
+      providers: c ? readAwsOrganizationConfig(c) : [],
+    });
+  }
+
+  constructor(options: {
+    providers: AwsOrganizationProviderConfig[];
+    logger: Logger;
+  }) {
+    this.providers = options.providers;
+    this.logger = options.logger;
+    let credentials = undefined;
+    if (
+      this.providers.length > 0 &&
+      this.providers[0].roleArn !== undefined &&
```

Comment:
yeah it totally sounds like you just wanted a single key rather than a providers array right